analyses/molecular-subtyping-MB/05-subtype-mb-shh.R

```diff
@@ -0,0 +1,275 @@
+
+# Create subtypes df that:
+# 1) calculates average methylations score per match id
+# 2) redefines SHH subtypes as alpha (3), beta (1), gamma (2), delta (4)
+# 3) append molecular alterations data
+# 4) attempts to classify samples without methylation classifiers into one of the four SHH subtypes using molecular and age criteria 
+subtypes <- pbta_mb_shh %>%
+  select(Kids_First_Participant_ID, sample_id, match_id, age_at_diagnosis_years, molecular_subtype, dkfz_v12_methylation_subclass, dkfz_v12_methylation_subclass_score) %>%
+  unique() %>%
+  group_by(Kids_First_Participant_ID, sample_id, match_id, age_at_diagnosis_years, molecular_subtype) %>%
+  summarize(
+    dkfz_v12_methylation_subclass_collapsed = if(all(is.na(dkfz_v12_methylation_subclass))) NA_character_ 
+    else paste(unique(na.omit(dkfz_v12_methylation_subclass)), collapse = "; "),
+    dkfz_v12_methylation_subclass_score_mean = mean(dkfz_v12_methylation_subclass_score, na.rm = TRUE),
+    .groups = "drop"
+  ) %>%
+  mutate(across(where(is.numeric), ~replace(., is.nan(.), NA)),
+         SHH_subtype = case_when(dkfz_v12_methylation_subclass_score_mean >= 0.7 & dkfz_v12_methylation_subclass_collapsed == "MB_SHH_1" ~ "SHH_beta",
+                                 dkfz_v12_methylation_subclass_score_mean >= 0.7 & dkfz_v12_methylation_subclass_collapsed == "MB_SHH_2" ~ "SHH_gamma",
+                                 dkfz_v12_methylation_subclass_score_mean >= 0.7 & dkfz_v12_methylation_subclass_collapsed == "MB_SHH_3" ~ "SHH_alpha",
+                                 dkfz_v12_methylation_subclass_score_mean >= 0.7 & dkfz_v12_methylation_subclass_collapsed == "MB_SHH_4" ~ "SHH_delta",
+                                 TRUE ~ NA_character_)) %>%
+  left_join(mb_shh_data_collapsed) %>%
+  left_join(germline) %>%
+  left_join(expr_df) %>%
+  unique() %>%
+  select(-match_id) %>%
+  # clean up subclass and scores
+  mutate(dkfz_v12_methylation_subclass_score_mean = case_when(dkfz_v12_methylation_subclass_score_mean < 0.7 & 
+                                                                !grepl("SHH", dkfz_v12_methylation_subclass_collapsed) ~ NA_integer_,
+                                                              TRUE ~ dkfz_v12_methylation_subclass_score_mean),
+         dkfz_v12_methylation_subclass_collapsed = case_when((dkfz_v12_methylation_subclass_score_mean < 0.7 | is.na(dkfz_v12_methylation_subclass_score_mean)) & 
+                                                               !grepl("SHH", dkfz_v12_methylation_subclass_collapsed) ~ NA_character_,
+                                                             TRUE ~ dkfz_v12_methylation_subclass_collapsed)) %>%
+  mutate(final_shh_subtype = case_when(
+                                        # For SHH alpha: >3 years AND (MYCN OR GLI2 amplification OR TPM>=2 OR chr17p loss)
+                                        SHH_subtype == "SHH_alpha" |
+                                         (age_at_diagnosis_years > 3 &
```

Comment:
```suggestion
                                         (age_at_diagnosis_years >= 2 &
```
I see this one 2.78 years old C4105494-Tumor which I think should be in SHH alpha due to MYCN high. I think maybe we go +/- 1 on age since these are so close.